### ISSUE TITLE
Error with VimDebuggerLoad and snipmate

### ISSUE THREAD

**Issue by jacobsandlund** [spf13/spf13-vim#27]
Launching MacVim through the terminal (mvim) shows these error messages.
I'm not sure how to fix them.

```
Error detected while processing function <SNR>15_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>15_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
Error detected while processing /Users/jtsandlund/dev/spf13-vim/.vim/bundle/snipmate/plugin/snipMate.vim:
line   31:
E117: Unknown function: funcref#Function
E116: Invalid arguments for function get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))
E15: Invalid expression: get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))
```

**Comment by spf13** [comment]
I believe [this patch](https://github.com/spf13/spf13-vim/commit/a8b74d43e684e35c7f5ed48cbddb17fd67f03d82) resolves this problem. Please pull and confirm.

**Comment by ngaloppo** [comment]
Interestingly, I'm getting a similar error, _after_ applying the patch. (this is on OSX FWIW)

```
nico@chocomousse:~/spf13-vim (master)$ Error detected while processing function <SNR>15_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>15_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
Error detected while processing function <SNR>15_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>15_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
```

**Comment by Andmat7** [comment]
After applying the patch,I'm getting a similar error, (I'ḿ using Ubuntu 11.04)

Se ha detectado un error al procesar function <SNR>13_VimDebuggerLoad:
línea    2
E117: Función desconocida: <SNR>13_VimDebuggerInit
Se ha detectado un error al procesar function DefPython:
línea  550
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
NameError: global name 'VimDebugger' is not defined

Tnanks

**Comment by fjk** [comment]
in VimDegugger.vim: Just switching the position of function 

function s:VimDebuggerLoad() ... and call s:VimDebuggerLoad() 
below the function s:VimDebuggerInit() ...
works fine for me 

fk

**Comment by bjartek** [comment]
fjk: your sollution also works for me to fix this bug.

**Comment by vertex** [comment]
fjk's solution worked for me. thanks

**Comment by lxyu** [comment]
do you mean change it to:

``` vim
function s:VimDebuggerInit()
function s:VimDebuggerLoad()
  if has('python')
    call s:VimDebuggerInit()
  endif
endfunction
call s:VimDebuggerLoad()
```

I still get this error:

```
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
Press ENTER or type command to continue
```

**Comment by pythonsnake** [comment]
@lxyu
Just move line 41 "call s:VimDebuggerLoad()" to line 77, between "endfunction" and "function! g:__dbg_WatchFoldText()"

**Comment by lxyu** [comment]
It works! Thank you~

**Comment by pythonsnake** [comment]
@lxyu
You're welcome

**Comment by wontondon** [comment]
fjk's fix worked for me as well

**Comment by danybmx** [comment]
thaks @pythonsnake it works for me! :)

**Comment by pythonsnake** [comment]
@danybmx
Cool.
You should thank fjk, the solution is from him. I just explained it :-)

**Comment by khuevu** [comment]
Hi guys, 
I am using the latest version of spf13-vim and I still have the error: 
Error detected while processing function <SNR>13_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>13_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
NameError: global name 'VimDebugger' is not defined
Anyone can help?

**Comment by samarpanda** [comment]
Open file ~/.vim/bundle/VimDebugger/plugin/VimDebugger.vim
Just move line 41 "call s:VimDebuggerLoad()" to line 77, between "endfunction" and "function! g:__dbg_WatchFoldText()"
Above fix worked for me.

Thanks.

**Comment by spf13** [comment]
fixed in the 3.0 release of spf13-vim

**Comment by samarpanda** [comment]
Great. Will upgrade to vim 3.0
Thanks.
